pkg/component/delete/delete.go

```diff
@@ -0,0 +1,30 @@
+package delete
+
+import (
+	"github.com/devfile/library/pkg/devfile/parser"
+	"github.com/redhat-developer/odo/pkg/component"
+	componentlabels "github.com/redhat-developer/odo/pkg/component/labels"
+	"github.com/redhat-developer/odo/pkg/kclient"
+	"github.com/redhat-developer/odo/pkg/libdevfile"
+)
+
+type DeleteComponentClient struct {
+	kubeClient kclient.ClientInterface
+}
+
+func NewDeleteComponentClient(kubeClient kclient.ClientInterface) *DeleteComponentClient {
+	return &DeleteComponentClient{
+		kubeClient: kubeClient,
+	}
+}
+
+func (o *DeleteComponentClient) UnDeploy(devfileObj parser.DevfileObj, path string) error {
+	undeployHandler := newUndeployHandler(path, o.kubeClient)
+	return libdevfile.Deploy(devfileObj, undeployHandler)
+}
+
+// DevfileComponentDelete deletes the devfile component
+func (do *DeleteComponentClient) DeleteComponent(devfileObj parser.DevfileObj, componentName string) error {
+	labels := componentlabels.GetLabels(componentName, "app", false)
+	return component.Delete(do.kubeClient, devfileObj, componentName, "app", labels, false, false)
```

Comment:
Do you want to move this package-level function `component.Delete` to this `delete` package? It would be the occasion to cleanup the `component` package, and to see what to do with the `spinner`: do we want to rely on spinner into this layer, or only on the CLI one?

Reply:
CLI does seem to be the ideal place to put spinner logic because it prints things to the terminal and is user facing, but then it is much more easier to place the spinner at the right place in the business logic.
But then we also discussed that business layer should just return an object and the CLI layer will print that object in JSON or human readable format.
I think we should stick to spinners in the CLI layer (write the methods in business logic in small pieces such that spinners at the CLI layer make sense), and use klog in the business layer for better understandably.

What do you think?